### PR TITLE
[test-11432][HttpTask] Add unit tests to http task

### LINKED ISSUE #11432: [Test][Http Task] Add unit tests to HttpTask
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Hi, i want add some unit tests to http task.

Mainly to test whether the HttpTask can execute the task normally, and whether the task status is correct under different HTTP check conditions.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by fangyinc**
## Purpose of the pull request

close [#11432](https://github.com/apache/dolphinscheduler/issues/11432)

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+ */
+
+package org.apache.dolphinscheduler.plugin.task.http;
+
+import static org.apache.dolphinscheduler.plugin.task.api.TaskConstants.EXIT_CODE_FAILURE;
+import static org.apache.dolphinscheduler.plugin.task.api.TaskConstants.EXIT_CODE_SUCCESS;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+import java.io.IOException;
+import java.util.*;
```
Comment (EricGao888) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: Please avoid importing *

Comment (fangyinc) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: Thanks, i will fix it

Comment (SbloodyS) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: > Please avoid importing *

I'm confuse why our spotless doesn't detect this. The previous checkstyle can be checked correctly.

Comment (EricGao888) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: > > Please avoid importing *
> 
> I'm confuse why our spotless doesn't detect this. The previous checkstyle can be checked correctly.

@SbloodyS I will look into it and give a quick fix for this.

Comment (fangyinc) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: @EricGao888 I have fixed this problem, please review again.

Comment (ruanwenjun) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: BTW, can we add `remove unused import` in spotless

Comment (EricGao888) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: > BTW, can we add `remove unused import` in spotless

@ruanwenjun Good idea! Actually I'm already working on the PR. Also we could add some extra rules for import order. Will submit a PR later tonight : )

File: dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+    @After
+    public void after() {
+        mockWebServers.forEach(IOUtils::closeQuietly);
+        mockWebServers.clear();
+    }
+
+    @Test
+    public void testHandleCheckCodeDefaultSuccess() throws Exception {
+        HttpTask getHttpTask = generateHttpTask(HttpMethod.GET, 200);
```
Comment (caishunfeng) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]:
```suggestion
        HttpTask getHttpTask = generateHttpTask(HttpMethod.GET, HttpStatus.SC_OK);
```

Comment (fangyinc) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-http/src/test/java/org/apache/dolphinscheduler/plugin/task/http/HttpTaskTest.java]: I have fixed this, please review again, thank you~

**Comment by fangyinc** [comment]
@EricGao888 @SbloodyS  Can someone review the code, if there is any problem I will fix it~

**Review by caishunfeng — APPROVED** [review]
LGTM

**Review by davidzollo — APPROVED** [review]
+1
good job

**Pull request merged**